hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMDBCheckpointServletInodeBasedXfer.java

```diff
@@ -0,0 +1,365 @@
+          shouldContinue = writeDBToArchive(copyFiles, sstFilesToExclude, snapshotDbPath,
+              maxTotalSstSize, archiveOutputStream, tmpdir);
+        }
+
+        RocksDBCheckpointDiffer differ = getDbStore().getRocksDBCheckpointDiffer();
+        Path sstBackupDir = new File(differ.getSSTBackupDir()).toPath();
+        Path compactionLogDir = new File(differ.getCompactionLogDir()).toPath();
+
+        if (shouldContinue) {
+          shouldContinue = writeDBToArchive(copyFiles, sstFilesToExclude, sstBackupDir,
+              maxTotalSstSize, archiveOutputStream,  tmpdir);
+        }
+
+        if (shouldContinue) {
+          shouldContinue = writeDBToArchive(copyFiles, sstFilesToExclude, compactionLogDir,
+              maxTotalSstSize, archiveOutputStream,  tmpdir);
+        }
+        // Add paths of snapshot databases, SST backups, and compaction logs to visit list
+        allPathsToVisit.addAll(snapshotDbPaths);
+        allPathsToVisit.add(sstBackupDir);
+        allPathsToVisit.add(compactionLogDir);
+      }
+
+      if (shouldContinue) {
+        // we finished transferring files from snapshot DB's by now and
+        // this is the last step where we transfer the active om.db contents
+        checkpoint = getCheckpoint(tmpdir,true);
+        // unlimited files as we want the Active DB contents to be transferred in a single batch
+        maxTotalSstSize.set(Long.MAX_VALUE);
+        Path checkpointDir = checkpoint.getCheckpointLocation();
+        writeDBToArchive(copyFiles, sstFilesToExclude, checkpointDir,
+            maxTotalSstSize, archiveOutputStream, tmpdir);
+        allPathsToVisit.add(checkpointDir);
+        generateAndWriteHardlinkFile(checkpointDir, allPathsToVisit, archiveOutputStream);
+      }
+
+    } catch (Exception e) {
+      LOG.error("got exception writing to archive " + e);
+      throw e;
+    } finally {
+      cleanupCheckpoint(checkpoint);
+    }
+  }
+
+  private static void cleanupCheckpoint(DBCheckpoint checkpoint) {
+    if (checkpoint != null) {
+      try {
+        checkpoint.cleanupCheckpoint();
+      } catch (IOException e) {
+        LOG.error("Error trying to clean checkpoint at {} .",
+            checkpoint.getCheckpointLocation().toString());
+      }
+    }
+  }
+
+  /**
+   * Generates a file containing hardlink information and writes it to the archive.
+   *
+   * @param checkpointDir Directory containing the checkpoint
+   * @param allPathsToVisit Set of all paths to visit for hardlink information
+   * @param archiveOutputStream Archive output stream
+   * @throws IOException if an I/O error occurs
+   */
+  private static void generateAndWriteHardlinkFile(Path checkpointDir, Set<Path> allPathsToVisit,
+      ArchiveOutputStream<TarArchiveEntry> archiveOutputStream) throws IOException {
+    Path data = Files.createTempFile(DATA_PREFIX, DATA_SUFFIX);
+    Path metaDirPath = getMetaDirPath(checkpointDir);
+    int truncateLength = metaDirPath.toString().length() + 1;
```

Comment:
This relative path wont work for snapshots. You are supposed to get the location relative om data directory. So an sst file corresponding to AOS om.db would become om.db/1.sst .
An sst file corresponding to snapshot db should become db.checkpoints/om.db-<snapshotId>/1.sst